runtime/src/main/java/org/corfudb/runtime/view/metadata/Checkpoint.java

```diff
@@ -0,0 +1,14 @@
+package org.corfudb.runtime.view.metadata;
+
+import java.util.List;
+import java.util.UUID;
+
+/**
+ * This class defines metadata associated with a checkpoint
+ */
+public class Checkpoint {
+    private UUID checkpointStream;
+    private long startAddress;
+    private long endAddress;
+    List<Long> checkpointRecords;
```

Comment:
If we assume this list is 1,000,000 pointers, then it takes ~35mb to represent in memory (using an ArrayList) and only 4mb in its serialized form. Seems like a small overhead?